lock/lock_test.go

```diff
@@ -320,6 +321,48 @@ func TestLocker(t *testing.T) {
 				}
 			},
 		},
+		{
+			description: "acquiring a lock sets CreatedAt",
+			testFn: func(t *testing.T) {
+				ctx := context.Background()
+				duration := 5 * time.Minute
+				input := AcquireLockInput{
+					Key:           "key1",
+					Owner:         "owner1",
+					LeaseDuration: &duration,
+				}
+				now := time.Now()
+				l := AcquireAndValidate(ctx, t, locker, input)
+				assert.WithinDuration(t, now, l.CreatedAt, 5*time.Second)
+			},
+		},
+		{
+			description: "heartbeating a lock we own resets created at",
+			testFn: func(t *testing.T) {
+				ctx := context.Background()
+				duration := 5 * time.Minute
+				input := AcquireLockInput{
+					Key:           "key1",
+					Owner:         "owner1",
+					LeaseDuration: &duration,
+				}
+
+				AcquireAndValidate(ctx, t, locker, input)
+
+				lock := Lock{
+					Key:   "key1",
+					Owner: "owner1",
+				}
+
+				now := time.Now()
+				newLock, _ := locker.HeartbeatLock(ctx, HeartbeatLockInput{
+					OriginalLock:  lock,
+					LeaseDuration: &duration,
+				})
+
+				assert.WithinDuration(t, now, newLock.CreatedAt, 5*time.Second)
```

Comment:
My guess is that, with such a wide tolerance as `5*time.Second`, this test doesn't really demonstrate that `HeartbeatLock` actually renewed the `CreatedAt`. Maybe add a `time.Sleep(1 * time.Second)` and reduce the tolerance to `1 * time.Second` (so that, because of the sleep, the original lock creation will no longer be `WithinDuration`?